includes/externalstore/ExternalStoreDB.php

```diff
@@ -317,7 +317,7 @@ private function fetchBlob( $cluster, $id, $itemID ) {
 			[ 'blob_id' => $id ],
 			__METHOD__
 		);
-		if ( $ret === false ) {
+		if ( $ret === false && !$this->isReadOnly( $cluster ) ) {
```

Comment:
**note**: I don't think this is correct assumption. While in readonly mode, it's totally fine to fallback to the Primary DB connection for reads. The problem in this case is that we cannot fetch the information and even if we will prevent the "exception" from no access to primary DB it will fail later as the BlobsStorage is required to render anything. So, most likely we will not need to change there anything, though we need to verify how often it happens and most importantly if this is happening in RES datacenter